notebook/static/notebook/js/menubar.js

```diff
@@ -128,6 +128,142 @@ define([
 
         this._new_window(url);
     };
+    
+    MenuBar.prototype._nbconvert_upload_conf = function (download) {
+      
+      var body =  $('<div>');
+      var notebook_path = utils.encode_uri_components(this.notebook.notebook_path);
+
+      var create_json = function(notebook, config){
+        var json_to_pass = {
+          notebook: notebook.toJSON(),
+          config: config
+        };
+        return json_to_pass;
+      };
+
+      
+      var form  = $('<form>');
+      var fileinput = $('<input>')
+        .attr('type', 'file')
+        .attr('tabindex', '1');
+
+      var fileformat = $('<select>');
+      
+      var export_opts = {
+        markdown: {
+          display_text:'markdown',
+          exporter_name:'markdown' 
+          },
+        html: {
+          display_text:'html',
+          exporter_name:'html' 
+          },
+        rst: {
+          display_text:'rst',
+          exporter_name:'rst' 
+          },
+        pdf: {
+          display_text:'PDF',
+          exporter_name:'pdf' 
+          },
+        latex: {
+          display_text:'LaTeX',
+          exporter_name:'latex' 
+          },
+        script: {
+          display_text:'script',
+          exporter_name:'script' 
+          }
+      };
+
+      for(var x in export_opts){
+        fileformat.append(
+          $('<option/>') 
+            .attr('value', export_opts[x]['exporter_name'])
+            .text(export_opts[x]['display_text'])
+        );
+      };
+      
+      form.append(fileformat);
+      form.append(fileinput);
+      body.append(form);
+      
+      var that = this;
+
+      var handle_nbconvert_post = function(){
+        var filereader = new FileReader();
+        filereader.onload = function(){
+          var my_config_data = filereader.result;
+          that['json_content'] = create_json(that.notebook, my_config_data);
+          on_done();
+        };
+        if (fileinput[0].files.length>0){
+          filereader.readAsText(fileinput[0].files[0]);
+        } else{
+          that['json_content']=create_json(that.notebook, '{}')
+          on_done();
+        }
+      };
+      var on_done = function(){
+        var url = utils.url_path_join(
+            that.base_url,
+            'nbconvert',
+            fileformat.val(),
+            notebook_path
+        ) + "?download=" + download.toString();
+          
+        var xsrf_token = utils._get_cookie('_xsrf');
+        var xhr = new XMLHttpRequest();
+        xhr.open('POST', url, true);
+        xhr.responseType = 'blob';
+        xhr.withCredentials = true;
+        xhr.setRequestHeader('X-XSRFToken', xsrf_token);
+        xhr.onreadystatechange = function(){
+          if(xhr.readyState === 4){
+            var blob = xhr.response;
+            var content_disposition = xhr.getResponseHeader('Content-Disposition');
+            var filename = content_disposition.match(/filename="(.+)"/)[1];
```

Comment:
Probably an error, it's just best for it to be a clear one rather than a cryptic one because something you assumed would always be true no longer is.